PSCoreScripts/Invoke-SmokeTestOnWebApp.ps1

```diff
@@ -0,0 +1,121 @@
+﻿<#
+.SYNOPSIS
+Perform a smoke test on a given web app
+
+.DESCRIPTION
+This script performs a smoke test on the URL a web app/function app is hosted at, appending a specified path onto the url.
+
+.PARAMETER AppName
+The name of the web/function app to smoke test
+
+.PARAMETER ResourceGroup
+The resource group the web app is in
+
+.PARAMETER Slot
+The slot to test.
+Optional, defaults to "production"
+
+.PARAMETER Path
+The path to smoke test
+
+.PARAMETER TimeoutInSecs
+A timeout for the smoke test requests.
+Optional, defaults to 5 seconds
+
+.PARAMETER BackOffPeriodInSecs
+A backoff period in between smoke test requests.
+Optional, defaults to 10 seconds
+
+.PARAMETER AttemptsBeforeFailure
+The number of attempts before the smoke test is marked as failing.
+Optiona, defaults to 5
+
+.EXAMPLE
+./Invoke-SmokeTestOnWebApp -AppName SomeWebApp -ResourceGroup SomeResourceGroup -Path /PathToTest
+
+#>
+
+param(
+    [Parameter(Mandatory=$true)]
+    [string] $AppName,
+    [Parameter(Mandatory=$true)]
+    [string] $ResourceGroup,
+    [Parameter(Mandatory=$false)]
+    [string] $Slot = "production",
+    [Parameter(Mandatory=$true)]
+    [string] $Path,
+    [Parameter(Mandatory=$false)]
+    [int] $TimeoutInSecs=5,
+    [Parameter(Mandatory=$false)]
+    [int] $BackOffPeriodInSecs=10,
+    [Parameter(Mandatory=$false)]
+    [int] $AttemptsBeforeFailure=5
+)
+
+function Invoke-SingleSmokeTest
+{
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Url,
+        [Parameter(Mandatory=$true)]
+        [int] $TimeoutInSecs
+    )
+
+    try {
+        Write-Verbose "Performing Invoke-WebRequest on Uri '$Url'.."
+        $result = Invoke-WebRequest -Method Get -Uri $Url -MaximumRedirection 0 -TimeoutSec $TimeoutInSecs -UseBasicParsing
+        return $result.StatusCode -eq 200
```

Comment:
Are we only considering 200s as a success or should other 2xx codes be included?

Reply:
Might be worth being explicit about what is considered a successful smoke test in the description

Reply:
I've expanded this to be any 2xx status code, and expanded the documentation to explicitly say this, and to say that it does not follow any redirects.